sympy/physics/continuum_mechanics/beam.py

```diff
@@ -293,49 +298,53 @@ def load(self):
         """
         return self._load
 
-    def solve_for_reaction_loads(self, *reactions):
-        """
-        Solves for the reaction forces.
-
-        Examples
-        ========
-        There is a beam of length 30 meters. A moment of magnitude 120 Nm is
-        applied in the clockwise direction at the end of the beam. A pointload
-        of magnitude 8 N is applied from the top of the beam at the starting
-        point. There are two simple supports below the beam. One at the end
-        and another one at a distance of 10 meters from the start. The
-        deflection is restricted at both the supports.
-
-        Using the sign convention of upward forces and clockwise moment
-        being positive.
-
-        >>> from sympy.physics.continuum_mechanics.beam import Beam
-        >>> from sympy import symbols, linsolve, limit
-        >>> E, I = symbols('E, I')
-        >>> R1, R2 = symbols('R1, R2')
-        >>> b = Beam(30, E, I)
-        >>> b.apply_load(-8, 0, -1)
-        >>> b.apply_load(R1, 10, -1)  # Reaction force at x = 10
-        >>> b.apply_load(R2, 30, -1)  # Reaction force at x = 30
-        >>> b.apply_load(120, 30, -2)
-        >>> b.bc_deflection = [(10, 0), (30, 0)]
-        >>> b.load
-        R1*SingularityFunction(x, 10, -1) + R2*SingularityFunction(x, 30, -1)
-            - 8*SingularityFunction(x, 0, -1) + 120*SingularityFunction(x, 30, -2)
-        >>> b.solve_for_reaction_loads(R1, R2)
-        >>> b.reaction_loads
-        {R1: 6, R2: 2}
-        >>> b.load
-        -8*SingularityFunction(x, 0, -1) + 6*SingularityFunction(x, 10, -1)
-            + 120*SingularityFunction(x, 30, -2) + 2*SingularityFunction(x, 30, -1)
-        """
+    def solve(self):
+        
         x = self.variable
         l = self.length
+        E = self.elastic_modulus
+        I = self.second_moment
+        C3 = self.C3
+        C4 = self.C4
         shear_curve = limit(self.shear_force(), x, l)
         moment_curve = limit(self.bending_moment(), x, l)
-        reaction_values = linsolve([shear_curve, moment_curve], reactions).args
-        self._reaction_loads = dict(zip(reactions, reaction_values[0]))
-        self._load = self._load.subs(self._reaction_loads)
+
+        slope_eqs = []
+        deflection_eqs = []
+
+        slope_curve = integrate(self.bending_moment(), x) + C3
+    
+        if not self._boundary_conditions['slope']:
+            #slope_curve = diff(self.deflection(), x)
+            pass
+        
+        else:
+            for position, value in self._boundary_conditions['slope']:
+                eqs = sympify(slope_curve.subs(x, position) - value)
+                slope_eqs.append(eqs)
+
+        if not self._boundary_conditions['deflection']:
+            pass
+
+        else:
+            deflection_curve = integrate(slope_curve, x) + C4
+
+            for position, value in self._boundary_conditions['deflection']:
```

Comment:
I think there is no need to keep this loop inside `if-else` block. If `not self._boundary_conditions['deflection']` is true then the loop would never be used.
Same goes for above loop.